### PR TITLE
cli: add a separate --domainname flag

### PR DESCRIPTION
**Pull request by cyphar**
A while ago, Docker split the "Domainname" field out from the "Hostname"
field for the container configuration. There was no real user-visible
change associated with this (and under the hood "Domainname" was mostly
left unused from the command-line point of view). We now add this flag
in order to match other proposed changes to allow for setting the NIS
domainname of a container.

[Cute cat by .:I&#x27;m-a-kitty-cat!:.](https://flic.kr/p/5ZniHY)

###### [Cute cat](https://flic.kr/p/5ZniHY) by [.:I&#x27;m-a-kitty-cat!:.](https://www.flickr.com/photos/tiina93/)

Signed-off-by: Aleksa Sarai <asarai@suse.de>

### CODE REVIEW

**Comment by cyphar** [comment]
This is still WIP because I haven't included any documentation changes.

It's also related to moby/moby#37302.

**Comment by cyphar** [comment]
This should be ready to review. It has tests and documentation (though it depends somewhat on the server-side changes as well).

**Comment by cyphar** [comment]
Alright, I've fixed the build by removing the Swarm changes -- it looks like Swarm doesn't have a Domainname field in the API so we can't really add a switch for it.

/ping @vdemeester @thaJeztah

**Comment by cyphar** [comment]
docker/docker#37302 was merged, so this should be ready-to-go.

File: contrib/completion/bash/docker [hunk 1]
```diff
@@ -3357,6 +3358,7 @@ _docker_service_update() {
 _docker_service_update_and_create() {
 	local options_with_args="
 		--endpoint-mode
+		--domainname
```
[the remark this answers is not in the document]
Comment (albers) [reply contrib/completion/bash/docker]: Adding the option here means that this option also valid for `service update`.
If this is correct, please add it to the docs for `service update` as well.

[the remark this answers is not in the document]
Comment (cyphar) [reply contrib/completion/bash/docker]: It shouldn't be available for `service <>` because the swarm API doesn't have a `Domainname` field. I'll remove it from auto-complete.

[the remark this answers is not in the document]
Comment (albers) [reply contrib/completion/bash/docker]: Please also remove the option from [the docs](https://github.com/docker/cli/pull/1130/files#diff-d4779426535b6679e406cf721c9f9289).

[the remark this answers is not in the document]
Comment (cyphar) [reply contrib/completion/bash/docker]: Fixed.

File: docs/reference/commandline/service_create.md [hunk 2]
```diff
@@ -758,7 +758,7 @@ The swarm extends my-network to each node running the service.
 Containers on the same network can access each other using
 [service discovery](https://docs.docker.com/engine/swarm/networking/#use-swarm-mode-service-discovery).
 
-Long form syntax of `--network` allows to specify list of aliases and driver options:  
+Long form syntax of `--network` allows to specify list of aliases and driver options:
```
Comment (albers) [inline docs/reference/commandline/service_create.md]: Please exclude this file from the commit. It only contains unrelated formatting changes.

Comment (cyphar) [reply docs/reference/commandline/service_create.md]: Fixed.

**Comment by cyphar** [comment]
@albers Removed the formatting change.

**Comment by albers** [comment]
Thanks for the contribution, @cyphar!
